spec/models/account_domain_block_spec.rb

```diff
@@ -3,22 +3,30 @@
 require 'rails_helper'
 
 RSpec.describe AccountDomainBlock do
+  let(:account) { Fabricate(:account) }
+
   it 'removes blocking cache after creation' do
-    account = Fabricate(:account)
     Rails.cache.write("exclude_domains_for:#{account.id}", 'a.domain.already.blocked')
 
-    described_class.create!(account: account, domain: 'a.domain.blocked.later')
-
-    expect(Rails.cache.exist?("exclude_domains_for:#{account.id}")).to be false
+    expect { build_account_domain_block('a.domain.blocked.later') }
+      .to change { account_has_exclude_domains_cache? }.to(false)
   end
 
   it 'removes blocking cache after destruction' do
-    account = Fabricate(:account)
-    block = described_class.create!(account: account, domain: 'domain')
+    block = build_account_domain_block('domain')
     Rails.cache.write("exclude_domains_for:#{account.id}", 'domain')
 
-    block.destroy!
+    expect { block.destroy! }
+      .to change { account_has_exclude_domains_cache? }.to(false)
+  end
+
+  private
+
+  def build_account_domain_block(domain)
```

Comment:
I think the `build` name may be a bit misleading, as in Rails parlance it implies the object is not persisted to the database. Maybe something like `block_domain_from_account` would be a better name? Alternatively, skipping this refactor (using `Fabricate` directly in call sites) is also a possibility.

Reply:
Good idea - updated to remove `build_` from name.